Source/Particles/Filter/FilterFunctors.H

```diff
@@ -0,0 +1,123 @@
+/* Copyright 2016-2020 Maxence Thevenet, Yinjian Zhao
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef FILTERFUNCTORS_H
+#define FILTERFUNCTORS_H
+
+#include "AMReX_Gpu.H"
+#include "AMReX_Random.H"
+#include "WarpXParticleContainer.H"
+#include "WarpX.H"
+
+using SuperParticleType = typename WarpXParticleContainer::SuperParticleType;
+
+/**
+ * \brief Functor that returns 0 or 1 depending on a random draw per particle
+ */
+struct RandomFilter
+{
+    /** constructor
+     * \param a_is_active whether the test is active
+     * \param a_fraction fraction of particles to select
+     */
+    RandomFilter(bool a_is_active, amrex::Real a_fraction)
+        : m_is_active(a_is_active), m_fraction(a_fraction) {};
+
+    /**
+     * \brief draw random number, return 1 if number < m_fraction, 1 otherwise
+     * \param p one particle
+     * \return whether or not the particle is selected
+     */
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    bool operator () (const SuperParticleType& p) const noexcept
+    {
+        if ( !m_is_active )
+        { return 1; }
```

Comment:
This is not a standard indentation. We use usually Allman or K&R, see https://en.wikipedia.org/wiki/Indentation_style and it **does make a difference to read code easily!** Please modify existing code only if there's a good reason to do so :).
From best to worst (IMHO):
```c++
        if ( !m_is_active ) return 1;

        if ( !m_is_active )
            return 1;

        if ( !m_is_active ){
            return 1;
        }

        if ( !m_is_active )
        {
            return 1;
        }
```